src/simularium/RemoteSimulator.ts

```diff
@@ -255,53 +260,52 @@ export class RemoteSimulator implements ISimulator {
         return `wss://${this.serverIp}:${this.serverPort}/`;
     }
 
-    private connectToUriAsync(address): Promise<string> {
-        const connectPromise = new Promise<string>((resolve) => {
-            this.connectToUri(address);
-            resolve("Successfully connected to uri!");
-        });
-
-        return connectPromise;
-    }
-
-    public connectToRemoteServer(address: string): Promise<string> {
+    public connectToRemoteServer(
+        address: string,
+        timeout = 1000
+    ): Promise<string> {
         const remoteStartPromise = new Promise<string>((resolve, reject) => {
+            const MAX_WAIT_TIME = 4 * timeout;
+            const MAX_RETRIES = 1;
+            let timeWaited = 0;
+            let retries = 0;
+
             if (this.socketIsConnected()) {
-                return resolve("Remote sim successfully started");
+                return resolve(CONNECTION_SUCCESS_MSG);
             }
 
-            const startPromise = this.connectToUriAsync(address);
-            // wait 1 second for websocket to open
-            const waitForIsConnected = () =>
+            // Wait a specified time for websocket to open
+            const waitForWebSocket = () =>
                 new Promise((resolve) =>
                     setTimeout(() => {
                         resolve(this.socketIsConnected());
-                    }, 1000)
+                    }, timeout)
                 );
 
-            const handleReturn = async () => {
-                const isConnected = await waitForIsConnected();
-                secondsWaited++;
+            /*
+            Initially wait for a max wait time of MAX_WAIT_TIME, then retry connecting 
+            <MAX_RETRIES> time(s). In a retry, only wait for the amount of time 
+            specified as timeout. (Can reset timeWaited to 0 upon retry if we want to
+            wait for MAX_WAIT_TIME in a retry too)
```

Comment:
Not positive if this is what we want, but I didn't change it

Reply:
It looks to me like we are waiting for a total of MAX_WAIT_TIME over many possible connection retries.  I am led to think that because of line 293 timeWaited+=timeout.  If I am correct, the comment should say something like "wait timeout time, try to reconnect, until a total of MAX_WAIT_TIME has elapsed before giving up"

Reply:
That's what I thought/expected too at first, but if you look at line 296-297 it won't let you get to a connection retry (in line 299) until `timeWaited` is at least at `MAX_WAIT_TIME`, which is what (along with some console logs) led to my interpretation. But this function really needs a test so that we know for sure how long we're making the user wait and how many retries we're allowing. I tried to break it out into a class method so I can test it but it turned into a tangled mess, so I reverted it 😅 But seeing your comment makes me want to try again, so I'll give it another go at testing this recursion.

Reply:
I see what you mean.  Thanks.  It's basically just polling on a shorter interval (`timeout`) but allowing for a long interval (MAX_WAIT_TIME) for the total time to connect.

Reply:
Yeah, I think the idea overall was that if it had failed to connect after waiting a while, waiting a second time wouldn't help much, the problem isn't a slow connection, but no connection. 

Have you tried to factor out `checkConnection` and pass in the two counters?

Reply:
That makes sense @meganrm . I factored out `checkConnection` but instead of passing in the counters as params I turned them into class properties. Let me know if there's a good reason to make them function params instead.